incubator/vault/templates/servicemonitor.yaml

```
@@ -0,0 +1,27 @@
+{{- if .Values.vaultExporter.serviceMonitor.enabled -}}
```

Similarly to my above comment, let's have this configuration be independent of how `vaultExporter` is implemented:

```suggestion
{{- if .Values.serviceMonitor.enabled -}}
```

Another chart reference: https://github.com/helm/charts/blob/e5e3be5b957b3373bd233c03431189bd9c5ce7a7/stable/envoy/values.yaml#L243